File: Strings/1446. Consecutive Characters/1446. Consecutive Characters.js
/**
 * @param {string} s
 * @return {number}
 */
var maxPower = function (s) {
  let max_power = 1;
  let curr_power = 1;

  for (let i = 1; i < s.length; i++) {
    if (s[i] == s[i - 1]) curr_power++;
    else {
      max_power = Math.max(max_power, curr_power);
      curr_power = 1;
    }
  }

  return Math.max(max_power, curr_power);
};

// Time: O(n)
// Space: O(1)

File: Strings/1446. Consecutive Characters/1446. Consecutive Characters.py
class Solution:
    def maxPower(self, s: str) -> int:
        max_power = 1
        curr_power = 1

        for i in range(1, len(s)):
            if s[i] == s[i - 1]:
                curr_power += 1
            else:
                max_power = max(max_power, curr_power)
                curr_power = 1

        return max(max_power, curr_power)

# Time: O(n)
# Space: O(1)